<launch>
  <!-- Define arguments -->
  <arg name="rgb_topic" default="/camera/color/image_raw" />
  <arg name="pose_topic" default="/franka_state_controller/O_T_EE" />
  <arg name="wrench_topic" default="/robotiq_force_torque_wrench_filtered" />


  <!-- Set parameters using arguments -->
  <param name="rgb_topic" value="$(arg rgb_topic)" />
  <param name="pose_topic" value="$(arg pose_topic)" />
  <param name="wrench_topic" value="$(arg wrench_topic)" />
  

  <!-- Nodes -->
  <!-- Camera Node launch -->
  <node pkg="visuomotor_policy_ros" type="gopro.py" name="gopro" output="screen" />

  <!-- Force sensor Node launch -->
  <node pkg="robotiq_force_torque_sensor_custom" type="rq_sensor_custom" name="rq_sensor_custom" output="screen" />
  <node pkg="robotiq_force_torque_sensor_custom" type="rq_stream_sensor_custom" name="rq_stream_sensor_custom" output="screen" />
  
  <!-- Franka Node launch -->
  <node pkg="visuomotor_policy_ros" type="online_running.py" name="online_running" output="screen">
    <param name="rgb_topic" value="$(arg rgb_topic)" />
    <param name="pose_topic" value="$(arg pose_topic)" />
    <param name="wrench_topic" value="$(arg wrench_topic)" />
  </node>
  
</launch>